src/pages/Home.vue

```diff
@@ -112,25 +106,56 @@ export default {
   computed: {
     /** Whether authentication method chosen is an ID entry */
     isAuthTypeID() {
-      return this.authType == 'ID';
+      return this.authType == "ID";
     },
+
     /** Whether authentication method chosen is OTP */
     isAuthTypeOTP() {
-      return this.authType == 'OTP';
+      return this.authType == "OTP";
     },
+
+    /** Checks if group exists */
     doesGroupExist() {
       return this.groupData;
     },
+
+    /** Retrieves destination platform */
+    getRedirectTo() {
+      return this.redirectTo == "" ? this.sessionData.redirectPlatform : this.redirectTo;
+    },
+
+    /** Retrieves destination ID */
+    getRedirectID() {
+      return this.redirectID == ""
+        ? this.sessionData.redirectPlatformParams.id
+        : this.redirectID;
+    },
+
+    /** Retrieves group name */
+    getGroup() {
+      return this.group == "" ? this.sessionData.group : this.group;
+    },
+
+    /** Returns the purpose value */
+    getPurpose() {
+      return this.purpose == "" ? this.sessionData.purpose : this.purpose;
+    },
+
+    /** Returns the purpose params  */
+    getPurposeParams() {
+      return this.purposeParams == ""
+        ? this.sessionData.purposeParams
+        : this.purposeParams;
+    },
   },
   async created() {
     /** If sessionId exists in route, then retrieve session details. Otherwise, fallback to using group data. */
-    if (this.sessionId == null) {
-      this.groupData = await groupAPIService.getGroupData(this.group);
-    } else {
+    if (this.sessionId != null) {
       this.sessionData = await sessionAPIService.getSessionData(this.sessionId);
-      this.sessionEnabled = (this.sessionData.enabled == 1);
+      this.sessionEnabled = this.sessionData.sessionActive;
     }
-    this.isLoading = false
+    this.groupData = await groupAPIService.getGroupData(this.group);
+    this.isLoading = false;
```

Comment:
in the earlier code version, one of two things was happening
Either we were getting group data or we were getting session data. But now both things are happening.
I didn't understand, why is that?

Reply:
In the earlier code, both were happening, it was just a difference in which component. 
So the flow is this: If no `sessionId` is mentioned, then get `group` from URL and fetch group details. If `sessionId` is in the URL, then fetch the session details first, compute if the session is active, then fetch group details to show the UI. 

If `sessionId` is in the URL, the group details were fetched in the SessionEntry component. This time, it's fetched in the Home component itself. Group API is called regardless.